ding/utils/deprecation.py

```diff
@@ -0,0 +1,57 @@
+import functools
+import textwrap
+import warnings
+
+
+def deprecated(since, removed_in, up_to=None):
+    """Decorate a function to signify its deprecation.
+
+    Args:
+        since: The version when the function was first deprecated.
+        removed_in: The version when the function will be removed.
+        up_to: The new API users should use.
+    """
+
+    def decorator(func):
+        existing_docstring = func.__doc__ or ""
+
+        deprecated_doc = f'.. deprecated:: {since}\n    This will be removed in {removed_in}'
+
+        if up_to is not None:
+            deprecated_doc += f', please use `{up_to}` instead.'
+        else:
+            deprecated_doc += '.'
+
+        # split docstring at first occurrence of newline
+        summary_and_body = existing_docstring.split("\n", 1)
```

Comment:
why split the string at the first occurrence of newline, the format of our comments in this project is like:
```python
def func(args1, args2):
    """
    Overview:
        This is a overview.
    Arguments:
        ...
    Returns:
        ...
    """
    ...
```